packages/react-storage/src/components/StorageBrowser/views/LocationActionView/hooks/useDeleteActionView.tsx

```diff
@@ -0,0 +1,219 @@
+import { humanFileSize, isFunction } from '@aws-amplify/ui';
+
+import { deleteHandler } from '../../../actions/handlers';
+import { TaskCounts } from '../../../controls/types';
+import {
+  DataTableProps,
+  DataTableRow,
+} from '../../../composables/DataTable/DataTable';
+import { WithKey } from '../../../components/types';
+import { getTaskCounts } from '../../../controls/getTaskCounts';
+import { STATUS_DISPLAY_VALUES } from '../../LocationActionView/constants';
+import { useStore } from '../../../providers/store';
+import { useGetActionInput } from '../../../providers/configuration';
+import { Task, useProcessTasks } from '../../../tasks';
+import { UseActionView } from './types';
+import {
+  getActionIconVariant,
+  getActionViewTaskStatuses,
+  getFilenameWithoutPrefix,
+  getFileTypeDisplayValue,
+} from './utils';
+
+interface UseDeleteActionView extends UseActionView {}
+
+const LOCATION_ACTION_VIEW_HEADERS: DataTableProps['headers'] = [
+  { key: 'key', type: 'sort', content: { label: 'Name' } },
+  { key: 'folder', type: 'text', content: { text: 'Folder' } },
+  { key: 'type', type: 'text', content: { text: 'Type' } },
+  { key: 'size', type: 'text', content: { text: 'Size' } },
+  { key: 'status', type: 'sort', content: { label: 'Status' } },
+  { key: 'action', type: 'text', content: { text: '' } },
+];
+
+export const getDeleteActionViewTableData = ({
+  tasks,
+  taskCounts,
+  path,
+}: {
+  tasks: Task[];
+  taskCounts: TaskCounts;
+  path: string;
+}): DataTableProps => {
+  const { hasStarted } = getActionViewTaskStatuses(taskCounts);
+
+  const rows: DataTableProps['rows'] = tasks.map((item) => {
+    const row: WithKey<DataTableRow> = {
+      key: item.key,
```

Comment:
I think @calebpollman may have a PR which adds UUIDs to these rows? We may be able just use those instead of the item.key (which is "probably" unique but the id would guarantee it)

Ditto for the cell keys

Reply:
updated